logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt

```diff
@@ -182,8 +185,47 @@ class ConversationDataSource(
      * Gets a flow that allows observing of
      */
     override suspend fun observeConversationDetailsById(conversationID: ConversationId): Flow<ConversationDetails> =
-        conversationDAO.observeGetConversationByQualifiedID(idMapper.toDaoModel(conversationID)).wrapStorageRequest().onlyRight()
-            .map(conversationMapper::fromDaoModel).flatMapLatest(::getConversationDetailsFlow)
+        conversationDAO.observeGetConversationByQualifiedID(idMapper.toDaoModel(conversationID))
+            .wrapStorageRequest()
+            .onlyRight()
+            .map(conversationMapper::fromDaoModel)
+            .flatMapLatest(::getConversationDetailsFlow)
+
+    private suspend fun getConversationDetailsFlow(conversation: Conversation): Flow<ConversationDetails> = when (conversation.type) {
```

Comment:
i think we have a convention that such funs should be named like `observeConversationDetails`

Reply:
I am not sure about this one as this function does not really observe anything